plugins/core/gs-appstream.c

```diff
@@ -1182,22 +1182,33 @@ gs_appstream_add_category_apps (GsPlugin *plugin,
 	}
 	for (guint j = 0; j < desktop_groups->len; j++) {
 		const gchar *desktop_group = g_ptr_array_index (desktop_groups, j);
-		g_autofree gchar *xpath = NULL;
+		g_autoptr(GString) xpath = g_string_new (NULL);
 		g_auto(GStrv) split = g_strsplit (desktop_group, "::", -1);
 		g_autoptr(GPtrArray) components = NULL;
 
 		/* generate query */
 		if (g_strv_length (split) == 1) {
-			xpath = g_strdup_printf ("components/component/categories/"
-						 "category[text()='%s']/../..",
-						 split[0]);
+			xb_string_append_union (xpath,
+						"components/component/categories/"
+						"category[text()='%s']/../..",
+						split[0]);
+			xb_string_append_union (xpath,
+						"component/categories/"
+						"category[text()='%s']/../../description/..",
```

Comment:
Changes lgtm to me but do we only want to select apps with description set?

Reply:
So, `<description>` from appdata acts as a differentiator to filter out non-applications.


A "real" system app in gnome-software contains 2 sources of metadata:
1) Desktop file (There is a XML-mapping (appstream-like) that gets pushed into the xmlb silo)
2) The appdata file.

Many non-application stuff have desktop files (for e.g. `ls  /usr/share/applications/ibus*`) 
We need to make sure this doesn't get bubbled up in the UI. Hence, check for the `<description>` tag which only is mentioned in the app's appdata. 

Indirectly, it also confirms that an desktop file has a real appdata file on  disk. I wrote about this at upstream places [here](https://gitlab.gnome.org/GNOME/gnome-software/issues/813) and [here](https://gitlab.gnome.org/GNOME/gnome-software/merge_requests/337)

Reply:
Keeping the conversation unresolved if you have more questions.

I think the PR is good to go, so merging it.
Thanks for the review